runtime/int.go

```diff
@@ -430,6 +441,20 @@ func intCheckedMul(v, w int) (int, bool) {
 	return x, true
 }
 
+func intCheckedPow(v, w int) (int, bool) {
+	// first try the shortcut - assuming the this operation is
```

Comment:
Strictly looking at math.Pow vs. float ** float, it seems like Go and Python are very similar:

```
Expression       Restriction                            Go       Python
Pow(x, ±0.0)                                            1.0      1.0
Pow(1.0, y)                                             1.0      1.0
Pow(x, 1.0)                                             x        x
Pow(NaN, y)                                             NaN      NaN
Pow(x, NaN)                                             NaN      NaN
Pow(±0.0, y)     y an odd integer < 0                   ±Inf     ZeroDivisionError
Pow(±0.0, y)     finite y < 0 and not an odd integer    +Inf     ZeroDivisionError
Pow(±0.0, y)     y an odd integer > 0                   ±0.0     ±0.0
Pow(±0.0, y)     finite y > 0 and not an odd integer    +0.0     +0.0
Pow(±0.0, -Inf)                                         +Inf     +Inf
Pow(±0.0, +Inf)                                         +0.0     +0.0
Pow(-1.0, ±Inf)                                         1.0      1.0
Pow(x, +Inf)     |x| > 1                                +Inf     +Inf
Pow(x, -Inf)     |x| > 1                                +0.0     +0.0
Pow(x, +Inf)     |x| < 1                                +0.0     +0.0
Pow(x, -Inf)     |x| < 1                                +Inf     +Inf
Pow(+Inf, y)     y > 0                                  +Inf     +Inf
Pow(+Inf, y)     y < 0                                  +0.0     +0.0
Pow(-Inf, y)     y an odd integer < 0                   -0.0     -0.0
Pow(-Inf, y)     finite y < 0 and not an odd integer    +0.0     +0.0
Pow(-Inf, y)     y an odd integer > 0                   -Inf     -Inf
Pow(-Inf, y)     finite y > 0 and not an odd integer    Inf      Inf
Pow(x, y)        finite x < 0 and finite non-integer y  NaN      ValueError
```